sdk/test-utils/recorder/src/recorder.ts

```diff
@@ -53,19 +58,58 @@ export class Recorder {
   private assetsJson?: string;
   private variables: Record<string, string>;
 
-  constructor(private testContext?: Test | undefined) {
+  constructor(
+    private testContext?:
+      | MochaTest
```

Comment:
We shouldn't hard code to either Mocha or Vitest here as we can just encompass them with interfaces.  See https://gist.github.com/mpodwysocki/c0afe1f92a16d839e4bc925cad647caf

Reply:
Switched to use your testInfo.ts.  I updated types for vitest to account for the nested suites.